massa-protocol-worker/src/handlers/block_handler/retrieval.rs

```diff
@@ -456,13 +456,39 @@ impl RetrievalThread {
             self.config.genesis_timestamp,
             slot,
         )?;
-        let version = self.mip_store.get_network_version_active_at(ts);
-        if header.content.current_version != version {
+        let current_version = self.mip_store.get_network_version_active_at(ts);
+        if header.content.current_version != current_version {
+            // Received a current version different from current version (given by mip store)
             Err(ProtocolError::IncompatibleNetworkVersion {
-                local: version,
+                local: current_version,
                 received: header.content.current_version,
             })
         } else {
+            if let Some(announced_version) = header.content.announced_version {
+                if announced_version <= current_version {
+                    // Received block, announced an old version (already accepted)
+                    return Err(ProtocolError::OutdatedAnnouncedNetworkVersion {
+                        local: current_version,
+                        announced_received: announced_version,
+                    });
+                }
+
+                match self.mip_store.get_network_version_to_announce() {
+                    Some(version_to_announce) => {
+                        // Unknown/new announced version -> user might need to upgrade
+                        if announced_version > version_to_announce {
+                            warn!("Please update your node if you want to support this update (new network version: {})", announced_version);
```

Comment:
we must not tell the node runner to update their node every time they receive a block announcing something unknown. The warning should only appear following the counters: eg. if more than 30% of the network is announcing something we don't know, then warn the user. Otherwise attackers can cause a lot of noise